schemas/user_tasks.proto

```diff
@@ -34,81 +62,170 @@ message UserTaskField {
   // 5. Default values and optional fields
 }
 
+// A UserTaskRun is a running instance of a UserTaskDef. It is created when a
+// ThreadRun arrives at a Node of type `USER_TASK`.
 message UserTaskRun {
+  // The ID of the UserTaskRun.
   UserTaskRunId id = 1;
+
+  // The ID of the UserTaskDef that this UserTaskRun comes from.
   UserTaskDefId user_task_def_id = 2;
 
+  // The user_group to which this UserTaskRun is assigned. Not Set if not assigned
+  // to a group. At least one of user_group or user_id will be set for any given
+  // UserTaskRun.
   optional string user_group = 3;
+
+  // The user_id to which this UserTaskRun is assigned. Not Set if not assigned
+  // to a user. At least one of user_group or user_id will be set for any given
+  // UserTaskRun. If user_id is set, then the UserTaskRun cannot be in the
+  // UNASSIGNED status.
   optional string user_id = 4;
 
+  // The results of the UserTaskRun. Empty if the UserTaskRun has not yet been completed.
+  // Each key in this map is the `name` of a corresponding `UserTaskField` on the
+  // UserTaskDef.
   map<string, VariableValue> results = 6;
+
+  // Status of the UserTaskRun. Can be UNASSIGNED, ASSIGNED, DONE, or CANCELLED.
```

Comment:
this is nice

Reply:
That was a team effort...we spent hours in design meetings trying to figure it out :sweat_smile: